### ISSUE TITLE
A newly created slot has a copy of all configuration settings from production slot

### ISSUE THREAD

**Issue by margani** [Azure/azure-cli#16805]
When I create a deployment slot, it has a copy of all the configuration settings from production slot, 

**Command Name**
```
az functionapp deployment slot create 
  --name MyFunctionApp
  --resource-group MyResourceGroup
  --slot staging
```

## To Reproduce:
Steps to reproduce the behavior.

- Create a function app in Azure
- Add an app setting or connection string, (no difference if it's a deployment slot settings).
- Create a new slot using:
`az functionapp deployment slot create --resource-group MyResourceGroup --name MyFunctionApp --slot staging`
or even using Azure Portal.

## Expected Behavior
I expect one of these:
- A created slot with no configuration copied from production or any other slot.
- A created slot with copied configuration settings which are not deployment slot settings in that slot.

I expect these because if I want to have a copy of configuration settings during slot creation, I could use [--configuration-source](https://docs.microsoft.com/en-us/cli/azure/functionapp/deployment/slot?view=azure-cli-latest#az_functionapp_deployment_slot_create-optional-parameters) parameter.

## Environment Summary
```
Windows-10-10.0.19041-SP0
Python 3.6.8
Installer: MSI

azure-cli 2.13.0 *

Extensions:
aks-preview 0.4.61
application-insights 0.1.1
azure-devops 0.13.0

```

**Comment by yonzhan** [comment]
functionapp

**Comment by naveedaz** [comment]
@panchagnula Can you please take a look if this is happening on the Az Cli or backend?

**Comment by panchagnula** [comment]
@naveedaz this is function app specific to tagging the right folks to look. @fabiocav - can someone from functions take a look.

**Comment by ulluoink** [comment]
@fabiocav @naveedaz @naveedaz @margani @yonzhan any updates on this? this is quite old but still happens!!!!!

**Comment by mlm-mjohnson** [comment]
I found this issue after trying to figure out why the config values weren't getting added when I specified a --configuration-source. After reading this thread I decided to leave  out the --configuration-source from my script and, sure enough, all the config from the parent resources was added to my slot. It seems the behavior is the opposite of what's expected.

TLDR: Adding --configuration-source results in no config, excluding --configuration-source results in all the config.
